### ISSUE TITLE
Left arrow resets view when navigating SI plots

### ISSUE THREAD

**Issue by thomasaarholt** [hyperspy/hyperspy#1866]
When zoomed in on an area in a plot, it is nice to be able to move forwards and around to compare neighbouring pixels using the arrow keys. When moving backwards (using the left arrow) however, this _also_ triggers the original behaviour of the left arrow in matplotlib displays, namely to reset the view to the previous view.

It would be nice to fully overwrite this so that left arrow only moves to the left, and does not reset the view.

**Comment by francisco-dlp** [comment]
A workaround for this is to use numbers (for convenience the ones in the numeric pad instead). However, it'll be nice to provide a real solution for this bug.

**Comment by thomasaarholt** [comment]
A possible solution is here: https://stackoverflow.com/questions/35624183/disable-matplotlibs-default-arrow-key-bindings. We should probably remove the forward (right arrow) behaviour as well as it acts as the opposite of the left arrow.

The suggested solution is rather brute force, but can be modified to be for a per axes basis:
```python
import matplotlib as mpl
mpl.rcParams['keymap.back'].remove('left')
mpl.rcParams['keymap.forward'].remove('right')
```
Edit: I've checked and the above does solve the problem.

**Comment by thomasaarholt** [comment]
Since lots of people use hyperspy on their laptop, I'd suggest adding *additional* support for the numpad.

**Comment by francisco-dlp** [comment]
Changing the default mpl keymap for the user's session after importing hyperspy sounds indeed like a brute force approach. What about doing it only when registering hspy keymap (i.e. on creating the multi dimensional signal figure) and restoring them when closing the figure?

**Comment by thomasaarholt** [comment]
Yes, that sounds good. I found people doing the following for temporarily setting the grid on, but I'm not sure how to remove things temporarily from a list:
```python
with plt.rc_context({"axes.grid": True, "grid.linewidth": 0.75}):
    f, ax = plt.subplots()  # Will make a figure with a grid
    ax.plot(x, y)
```
Could we do something along the lines of
```python
with plt.rc_context({"keymap.back": [],}):
   plot_stuff()
```

Edit: Nope, the above does not work.

**Comment by francisco-dlp** [comment]
We can't use a context manager here because the user starts interacting with the plot after the plot is finished.

I am wondering if it wouldn't be better to simply change hspy's keymap to ``Ctrl + Arrow``. 

Also, it would be useful to add a message to the log explaining the keymaps everytime  that a keymap is registered for a given plot. I've found advanced hyperspy users who don't know that you can navigate the signal in this way because they've never gone through that part of the user guide. This could also help making users aware of changes in keymaps if e.g. my proposal above is accepted.

**Comment by thomasaarholt** [comment]
What do you mean by "add a message to the log"? Do you mean a warning? I'd get a bit tired of seeing that, but I suppose we could set an option in the preferences.

**Comment by thomasaarholt** [comment]
Huh. I had not realised that you can use the numpad keys (with Num Lock on) for moving on the plot. That simplifies things. Unless there is disagreement, I'll add support for 3 and 4 navigation dimensions by adding  7,9 and 1,3 respectively for those dimensions. There have been times where it's been rather annoying to have to move using a slider in addition to arrows.

**Comment by francisco-dlp** [comment]
Adding support for extra navigation dimension would be great. What about:

* ``Ctrl + Arrows``: navigate navigation dimensions 1 and 2
* ``Ctrl + Shift + Arrows``: navigate navigation dimensions 3 and 4
* ``Ctrl + Alt + Arrows``: navigate navigation dimensions 5 and 6

**Comment by thomasaarholt** [comment]
Sounds good - let's do the same for the numpad.

What is the syntax for events when it comes to ctrl and alt? Currently it's
just 'left' etc.

**Comment by francisco-dlp** [comment]
https://matplotlib.org/api/backend_bases_api.html?highlight=key%20event%20modifiers#matplotlib.backend_bases.KeyEvent

**Comment by francisco-dlp** [comment]
Fixed in #1870
